recipes/moltenvk/all/conanfile.py

```diff
@@ -0,0 +1,182 @@
+            "1.0.35": "1.1.108.0",
+            "1.0.34": "1.1.106.0",
+            "1.0.33": "1.1.101.0",
+            "1.0.32": "1.1.97.0",
+            "1.0.31": "1.1.97.0",
+            "1.0.30": "1.1.92.0",
+            "1.0.29": "1.1.92.0",
+            "1.0.28": "1.1.92.0",
+            "1.0.27": "1.1.92.0",
+            "1.0.26": "1.1.85.0",
+            "1.0.25": "1.1.85.0",
+            "1.0.24": "1.1.85.0",
+            "1.0.23": "1.1.85.0",
+            "1.0.22": "1.1.82.0",
+            "1.0.21": "1.1.82.0",
+            "1.0.20": "1.1.82.0",
+            "1.0.19": "1.1.82.0",
+            "1.0.18": "1.1.82.0",
+            "1.0.17": "1.1.82.0",
+        }[self.version]
+
+    def package_id(self):
+        # MoltenVK >=1.O.42 requires at least XCode 12.0 (11.4 actually) at build
+        # time but can be consumed by older compiler versions
+        if tools.Version(self.version) >= "1.0.42":
+            if tools.Version(self.settings.compiler.version) < "12.0":
+                compatible_pkg = self.info.clone()
+                compatible_pkg.settings.compiler.version = "12.0"
+                self.compatible_packages.append(compatible_pkg)
+
+    def validate(self):
+        if tools.Version(self.version) >= "1.0.42":
+            if tools.Version(self.settings.compiler.version) < "12.0":
+                raise ConanInvalidConfiguration("MoltenVK {} requires XCode 12.0 or higher at build time".format(self.version))
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("MoltenVK-" + self.version, self._source_subfolder)
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        # Inject commit hash of spirv-cross
+        mvkdevice_mm = os.path.join(self._source_subfolder, "MoltenVK", "MoltenVK", "GPUObjects", "MVKDevice.mm")
+        if tools.Version(self.version) < "1.0.44":
+            tools.replace_in_file(mvkdevice_mm,
+                                  "#include <SPIRV-Cross/mvkSpirvCrossRevisionDerived.h>",
+                                  "static const char* spirvCrossRevisionString = \"{}\";".format(self._spirv_cross_commit_hash))
+        else:
+            tools.replace_in_file(mvkdevice_mm,
+                                  "#include \"mvkGitRevDerived.h\"",
+                                  "static const char* mvkRevString = \"{}\";".format(self._spirv_cross_commit_hash))
+
+    @property
+    def _spirv_cross_commit_hash(self):
+        return {
+            "20210115": "9acb9ec31f5a8ef80ea6b994bb77be787b08d3d1",
+            "20200917": "8891bd35120ca91c252a66ccfdc3f9a9d03c70cd",
+            "20200629": "b1082c10afe15eef03e8b12d66008388ce2a468c",
+            "20200519": "3c43f055df0d7b6948af64c825bf93beb8ab6418",
+            "20200403": "6637610b16aacfe43c77ad4060da62008a83cd12"
+        }[self.deps_cpp_info["spirv-cross"].version]
```

Comment:
I may remove this part and avoid the need to hardcode commit hash of each `spirv-cross` version. Hash will be different because almost none library in Vulkan ecosystem relies on tags of others repos, just some random commit between tags (lot of fun).

It requires to add a custom target in CMakeLists, which would:
- read spirv-cross commit hash from `ExternalRevisions/SPIRV-Cross_repo_revision` file (available in MoltenVK archive).
- write this information in the generated header file included by `MVKDevice.mm`